Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp

```diff
@@ -123,7 +122,55 @@ namespace ROS2::Utils
     }
 
 
+    AvailableAsset GetAvailableAssetInfo(const AZStd::string& globalSourceAssetPath)
+    {
+
+        using AssetSysReqBus = AzToolsFramework::AssetSystemRequestBus;
+        AvailableAsset foundAsset;
+
+        // get relativePath
+        bool relativePathFound{ false };
+        AZStd::string relativeSourcePath;
+        AZStd::string rootFilePath;
+        AssetSysReqBus::BroadcastResult(
+            relativePathFound, &AssetSysReqBus::Events::GenerateRelativeSourcePath,globalSourceAssetPath, relativeSourcePath, rootFilePath);
+        if (!relativePathFound){
+            return foundAsset;
+        }
+
+        // get source asset info
+        bool sourceAssetFound{ false };
+        AZ::Data::AssetInfo assetInfo;
+        AZStd::string watchFolder;
+        AssetSysReqBus::BroadcastResult(
+            sourceAssetFound, &AssetSysReqBus::Events::GetSourceInfoBySourcePath, relativeSourcePath.c_str(), assetInfo, watchFolder);
+        if (!sourceAssetFound){
+            return foundAsset;
+        }
 
+        foundAsset.m_sourceGuid = assetInfo.m_assetId.m_guid;
+
+        foundAsset.m_assetId = assetInfo.m_assetId;
```

Comment:
Beware of using this as an actual assetid.  The API is a bit confusing.

GetSourceInfoBySourcePath fills in the assetinfo structure, which sets the relative path, as well as the assetInfo.m_assetId, but it only sets the m_guid field of the assetId and always sets the subid to 0.  A real assetid is made of a tuple
```
  m_guid;  // the UUID of the source file (ie, the DAE or FBX or TIF or PNG or whatever)
  m_subId; // a unique number for each product generated by that source, since its 1-to-many
```

So when asking about a source file, using GetSourceInfoBySourcePath, the subid will always be 0 - this is not actually a valid assetid, its possible for a source file (like a TIF or DAE or something) to never output a product with subid 0 (its not an index, its a hash).

to get the subids (and thus the full assetid) you would need to query the output product(s) given that input source uuid, which is a different API which takes the source UUID (which can be gotten here, and also doesn't require use of GenerateRelativeSourcePath and also does not require you to use the database, its just another bus call, given a UUID of a source you can get all the output products from that source)

Reply:
Removed m_assetId from my helper structure here to not cause confusion. It is used only to track source files, and m_subId is never used. The previous code was caused by my lack of understanding. Thank you for the clarification! :)